### PR TITLE
Ipc task for 1 client

### PR DESCRIPTION
**Pull request by bogdann98**

### CODE REVIEW

File: bogdann98_task3/DynamicStringArray.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            if(String==dynamicArray[i])
+            {
+                found=1;
+                break;
+            }
+        }
+        if(found==1)
+        {
+            string *temp=dynamicArray;
+            dynamicArray=new string[size -1];
```
Comment (maxlig79) [inline bogdann98_task3/DynamicStringArray.cpp]: Looks like a memory leak

File: bogdann98_task3/DynamicStringArray.cpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            for(int j;j<size;j++)
```
Comment (maxlig79) [inline bogdann98_task3/DynamicStringArray.cpp]: What is j start value?

File: bogdann98_task3/DynamicStringArray.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        int i;
+        for(i=0;i<size;i++)
+        {
+            if(String==dynamicArray[i])
+            {
+                found=1;
+                break;
+            }
+        }
+        if(found==1)
```
Comment (maxlig79) [inline bogdann98_task3/DynamicStringArray.cpp]: found is a bool, please use true and false.

File: bogdann98_task3/DynamicStringArray.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    public:
+    DynamicStringArray();
+    int getSize() const;
+    void addEntry(const string &String);
+    bool deleteEntry(const string &String);
+    string *getEntry(const int index);  
+    DynamicStringArray(const vector<string> &vec);
+    DynamicStringArray(const DynamicStringArray &obj);
+    DynamicStringArray(DynamicStringArray &&obj);
```
Comment (maxlig79) [inline bogdann98_task3/DynamicStringArray.h:23]: I can't find the implementation of this constructor

File: bogdann98_task3/server.cpp [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  try
+  {
+      while(turn_on==true)
+      {
+        mq.receive(&mqrstruct,MAX_MESSAGE_SIZE,received_size,priority);
+          if(clientArray.find(mqrstruct.client) == clientArray.end())
+      {
+        clientArray.insert(std::make_pair(mqrstruct.client, std::make_unique<DynamicStringArray>() ) );
+      }
+        CommandPair commandPair=split_command(mqrstruct.command);
```
Comment (maxlig79) [inline bogdann98_task3/server.cpp]: Please do something with the indentation...

File: bogdann98_task3/DynamicStringArray.cpp [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    newdynamicArray[size] = String;
+    delete[] dynamicArray;
+    dynamicArray = newdynamicArray;
+    size++;
+}
+// the function boolean should search dynamicArray for the string
+bool DynamicStringArray::deleteEntry(const string &String)
+{
+    bool found = 0;
```
Comment (maxlig79) [inline bogdann98_task3/DynamicStringArray.cpp]: found is a bool or int?

File: bogdann98_task3/DynamicStringArray.cpp [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    if (dynamicArray[0] == String && size == 1)
+    {
+        delete[] dynamicArray;
+        size--;
+        found = 1;
```
Comment (maxlig79) [inline bogdann98_task3/DynamicStringArray.cpp]: bool or int?

File: bogdann98_task3/DynamicStringArray.cpp [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    if (dynamicArray[0] == String && size > 1)
```
Comment (maxlig79) [inline bogdann98_task3/DynamicStringArray.cpp]: So if the String is not found you will try anyway to modify the dynamicArray?
Can you please add a test where you delete a string that is not in the dynamicArray and see what happen.

**Review by maxlig79 — CHANGES_REQUESTED** [review]

**Review by maxlig79 — APPROVED** [review]

**Review by silvanadomnica — APPROVED** [review]

**Pull request merged**